### PR TITLE
libkb: optimization: only load necessary fields from user/lookup

### PR DESCRIPTION
**Pull request by maxtaco**
- strip out some dead code too

### CODE REVIEW

**Comment by maxtaco** [comment]
PTAL @patrickxb ?

**Comment by maxtaco** [comment]
It turns out there is also a security flaw here. The user shouldn't trust the server's output for sigs, which isn't trusted.  This PR fixes that.

File: go/libkb/kbsig.go [hunk 1]
```diff
@@ -137,7 +137,15 @@ func (u *User) ToTrackingStatementBasics(errp *error) *jsonw.Wrapper {
 }
 
 func (u *User) ToTrackingStatementSeqTail() *jsonw.Wrapper {
-	return u.sigs.AtKey("last")
```
Comment (maxtaco) [inline go/libkb/kbsig.go]: this isn't a good idea, since this dictionary arrives from the server, untrusted.

**Comment by patrickxb** [comment]
LGTM

**Pull request merged**